### DISCUSSION TITLE
vtk remote rendering, how wslink to close the launched process, how client communicate the server for user interaction

### DISCUSSION THREAD

**Discussion by guozhi72** [Kitware/trame#278] [General]
Hi, @sebastien:

Thanks for all help about the laucher for vtk remote rendering. 
For my development, I am running my client code using port 4200, the launcher is using port 8080 for rest api. So need to in client code, tell to connect to port 8080, also need to set up proxy for cross origin. Setting as shown below:
![setting](https://github.com/Kitware/trame/assets/84816693/45533917-aeb9-47ed-bb24-1ba918554ef0)
Then  I can do multiple client, multiple cotent rendeing.

![multiple content_muliple client](https://github.com/Kitware/trame/assets/84816693/8bb74399-0eff-48d4-8454-70056e1a867f).

But now, after I closed the page, the launcher did not stop the rending process. Thanks for all your help. Now I believe I set up the development environment, but new problem will be
1, how to close the render application when user close the browser?
2, As any visualization package, if user do some interactive interaction, server need to reponse? can you send me some link or example how to handler user interaction at client code, python server application code?

Best

**Comment by jourdain** [comment]
1. Wait 300s or adjust the `--timeout` of your processes. 
2. No idea of what you are asking. (Maybe you are asking for [that](https://github.com/Kitware/web-project-templates/tree/master/vue-vtkjs-pvw-template))

**Comment by jourdain** [comment]
Not trame related, closing...
